.github/workflows/build-test-publish-docker.yml

```diff
@@ -0,0 +1,74 @@
+name: build-test-publish-docker
+run-name: "Build, test, publish Docker images (branch ${{ github.ref_name }} on ${{ github.event_name }}/${{ github.event.action }} by ${{ github.actor }})"
+
+# This workflow runs the build, test and publish steps for the
+# Keyman build Docker images.
+# It is triggered by a schedule and repository_dispatch event.
+
+on:
+  schedule:
+    #run every Fri, 3 pm UTC
+    - cron: '0 15 * * 5' #
+
+  repository_dispatch:
+    types: ['build-test-publish-docker:*']
+
+env:
+  # Use GitHub registry, if empty
+  REGISTRY: ghcr.io
+  USERNAME: "${{ github.actor }}"
+  PASSWORD: "${{ github.token }}"
+
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Build Docker images
+        run: |
+          echo ${{ github.event.action }}
+          if [ -z "${{ env.USERNAME }}" ]; then
+            resources/docker-images/build.sh build: --registry ${{ env.REGISTRY }}
+          else
+            echo ${{ env.PASSWORD }} | resources/docker-images/build.sh build: --registry ${{ env.REGISTRY }} --username ${{ env.USERNAME }}
+          fi
+
+
+  test:
+    runs-on: ubuntu-latest
+    needs: build
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Test Docker images
+        run: |
+          if [ -z "${{ env.USERNAME }}" ]; then
+            resources/docker-images/build.sh test: --registry ${{ env.REGISTRY }}
+          else
+            echo ${{ env.PASSWORD }} | resources/docker-images/build.sh test: --registry ${{ env.REGISTRY }} --username ${{ env.USERNAME }}
```

Comment:
Won't this build the images again? It's running in a new container where it doesn't have the images available from the previous step!? Do we have to pass the artifacts between the steps?

(Similar for publish step)

Reply:
The `test:`step takes the images either locally or from the remote repository, it won't build the images again. 
There is probably another problem, the `publish:` step should be right after the `build:` step, before the `test:` step.